app/goal_routes.py

```diff
@@ -0,0 +1,156 @@
+from app.models.task import Task
+
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+def validate_complete_request(request_body):
+    try:
+        if request_body["title"]:
+            return request_body
+
+    except:
+        abort(make_response({"details": "Invalid data"}, 400))
+
+
+def validate_model_id(cls, model_id):
+    try:
+        model_id = int(model_id)    
+    except:
+        abort(make_response({"details": "Invalid data"}, 404))
+
+    model = cls.query.get(model_id)
+
+    if not model:
+        abort(make_response({"details": "Invalid data"}, 404))
+    
+    return model
+
+
+@goals_bp.route("", methods=["POST"])
+def create_goal():
+    request_body = request.get_json()
+    valid_data = validate_complete_request(request_body)
+    new_goal = Goal.from_dict(valid_data)
+
+    db.session.add(new_goal)
+    db.session.commit()
+
+    goal_response = {
+        "goal": new_goal.to_dict()
+    }
+    return make_response(jsonify(goal_response), 201)
+
+
+@goals_bp.route("", methods=["GET"])
+def get_all_goals_sort_asc():
+    goal_query = Goal.query.all()
+    title_query = request.args.get("title")
+    if title_query:
+        goal_query = Goal.query.filter_by(title=title_query)
+
+    goals = goal_query
+
+    goals_response = [goal.to_dict() for goal in goals]
+
+    return make_response(jsonify(goals_response), 200)
+
+
+@goals_bp.route("/<goal_id>", methods=["GET"])
+def get_one_goal(goal_id):
+    goal = validate_model_id(Goal, goal_id)
+
+    goal_response = {
+        "goal": goal.to_dict()
+    }
+
+    return make_response(jsonify(goal_response), 200)
+
+@goals_bp.route("/<goal_id>", methods=["PUT"])
+def goal_update_entire_entry(goal_id):
+    goal = validate_model_id(Goal, goal_id)
+    request_body = request.get_json()
+    goal.title = request_body["title"]
+
+    db.session.commit()
+
+    goal_response = {
+        "goal": goal.to_dict()
+    }
+
+    return make_response((goal_response), 200)
+
+
+@goals_bp.route("/<goal_id>", methods=["DELETE"])
+def goal_delete(goal_id):
+    goal = validate_model_id(Goal, goal_id)
+
+    db.session.delete(goal)
+    db.session.commit()
+
+    return make_response({'details': f'Goal {goal.id} "{goal.title}" successfully deleted'}, 200)
+
+@goals_bp.route("/<goal_id>/tasks", methods=["POST"])
+def add_tasks_to_goal(goal_id):
+    goal = validate_model_id(Goal, goal_id)
+    request_body = request.get_json()
+
+    for task_id in request_body["task_ids"]:
+        task = validate_model_id(Task, task_id)
+        goal.tasks.append(task)
+        db.session.commit()
+
+    goal_response = {
+        "id": goal.id,
+        "task_ids": request_body["task_ids"]
+    }
+    
+
+    return make_response(jsonify(goal_response), 200)
+
+
+@goals_bp.route("/<goal_id>/tasks", methods=["GET"])
+def get_goal_with_tasks(goal_id):
+    goal = validate_model_id(Goal, goal_id)
+
+    tasks = []
+    for task in goal.tasks:
+        tasks.append(
```

Comment:
Consider validating `task` before appending it to `tasks`, like: 
```python
for task in goal.tasks:
        task = validate_model_id(Task, task.id)
        tasks.append(task)
```

then you don't need to write out the dictionary literal that represents a task because validate_model_id returns an instance of task.